tests/test_utils.py

```diff
@@ -856,6 +856,27 @@ def test_csv_dialect_detection_semicolon_delimited(self):
         self.assertEqual(lookup_dict[4]['header3'], 'bar')
         self.assertEqual(lookup_dict[1]['header1'], 1)
 
+    def test_csv_utf8_encoding(self):
```

Comment:
This test looks identical to `test_csv_utf8_bom_encoding` unless I'm missing something subtle.

Reply:
Removed the comment on line 864.

Reply:
Just to confirm, defaulting to `utf-8-sig` will handle both cases, where the BOM is present and BOM is not present?  I think that's what I'm reading here, but wanted to check (fantastic if so!)

Reply:
Yep, it works in both cases. From https://docs.python.org/3/library/codecs.html#standard-encodings:
> On encoding the utf-8-sig codec will write 0xef, 0xbb, 0xbf as the first three bytes to the file. On decoding utf-8-sig will skip those three bytes if they appear as the first three bytes in the file.